### DISCUSSION TITLE
chapters/basic/install-python

### DISCUSSION THREAD

**Discussion by giscus** [kaochenlong/kaochenlong-discuss#6] [General]
# chapters/basic/install-python

這是一本新手也能看的懂的 Python 程式語言教課書。本書將涵蓋 Python 程式語言以及網站製作的基本概念，同時也提供程式碼範例，以便讀者實際操作和學習。學習不僅為公司或工作，更是為了自己的成長和樂趣。如果你想要學習 Python 這個程式語言或是用它製作網站，這本書將會是您的最佳起點。

https://pythonbook.cc/chapters/basic/install-python

**Comment by gbaian10** [comment]
補充個 poetry update 版本升級的一個小坑，對於主要版本號(Major)是 0 開頭的 package，有特殊的 update 規則。

^0.2.3 的升級範圍是 >=0.2.3 <0.3.0
^0.0.3	的升級範圍是 >=0.0.3 <0.0.4
如果想在0版內無限制升級套件則必須使用 ^0

詳細可以參考 [poetry 的文件說明](https://python-poetry.org/docs/dependency-specification/#version-constraints)，看文件的表真的比較清楚!

這是我實際遇過的小坑，想說怎 update 套件都升級不上去，難道是有版本衝突? 結果發現是0版的升級規則是不一樣的。
~~有些套件已經開發了好幾年，github 星星數也都破萬的，還是死都不把版本號提升到1以上~~

**Comment by kuanhua** [comment]
"如果要離開 REPL 環境，輸入 exit() 後按下 Enter 鍵或是按 Ctrl + D 組合鍵就可以離開。"  <- ... 按 Ctrl + Z 組合鍵...

**Reply by kaochenlong** [reply]
這是認真的嗎 :)

`Ctrl + Z` 組合會把應用程式壓到背景執行而已，不會真的退出喔 :)

**Reply by kuanhua** [reply]
補充，在Windows 下使用Ctrl+z。

**Comment by goodmanbuild** [comment]
我在安裝 pyenv-win 這邊遇到一些問題，後續參考這篇，按照步驟即可正常參考，提供給大夥
https://medium.com/@programman.go616/%E7%94%A8pyenv-win%E7%AE%A1%E7%90%86%E4%BD%A0%E7%9A%84python%E7%89%88%E6%9C%AC-10b5441bf57e

**Comment by Dylan0916** [comment]
執行 `$ poetry shell` 後遇到 "The command "shell" does not exist." 的問題，查了一下[看起來](https://python-poetry.org/docs/cli/#shell)被移到 [poetry-plugin-shell](https://github.com/python-poetry/poetry-plugin-shell) 去了，要額外安裝

**Reply by kaochenlong** [reply]
感謝提醒，看來是這樣沒錯 :)

**Comment by ixlin** [comment]
我在Mac OS 执行 $ python 命令时报错“zsh: command not found: python”，需要先根据自己的shell类型，执行下面两条命令：
echo 'eval "$(pyenv init -)"' >> ~/.zshrc
source ~/.zshrc
